docs/developers/talawa-api/error-handling.md

```diff
@@ -0,0 +1,1110 @@
+                        path: "UserInput.password"
+                        minimumLength: 8
+                   })
+           }
+           // Approach when we need to fall on a general error based on the interface contract `UserError`
+           If (CERTAIN CHECK WHERE WE WOULD NEED TO ADD THAT ERROR IN THE 
+                  signUpErrors ARRAY) {
+                    signUpErrors.push({
+                            __typename:"UserError" ,
+                            message: "message" , 
+                            path: "path"
+                    })
+                  
+            }
+
+          // Here we will be returning multiple errors in the form on an array of signUpErrors
+          if (signUpErrors) {
+               return  {
+                   signUpData: {
+                         user : userObj,
+                          .... other AuthData fields
+                   } , 
+                   signUpErrors
+
+               }
+          }
+          
+          EVERYTHING IS OKAY, CREATE THE USER IN DB,
+          
+          CreatedUserObj = DB.CREATE(USER);
+
+          return  {
+                   signUpData: {
+                         user : CreatedUserObj,
+                          .... other AuthData fields
+                   } , 
+                   signUpErrors
+
+           }
+
+     }
+  }
+}
+```
+
+In this approach for resolving Field Errors ->
+
+1. Multiple Errors can be sent back to the client when needed in case the client apps need to display those errors concurrently with UI elements
+2. A general purpose interface contract `UserError` to fall back to in case we want to send a general purpose error back to the client.
+
+
+#### Client
+
+Let us take a look at how the clients would be making this query.
+
+```gql
+mutation {
+  signUp(
+    input: {
+      firstName: "Harry"
+      lastName: "Potter"
+      email: "someDuplicateEmail"
+      password: "12345"
+    }
+    ) {
+    signUpData { 
+      user: {
+        _id,
+        firstName,
+        lastName,
+        email
+      } ,
+      accessToken,
+      refreshToken
+     }
+    signUpErrors {
+      # Specific cases
+      ... on EmailTaken {
+        __typename
+        message
+        path
+        suggestion
+      }
+
+      ... on PasswordTooShort {
+        __typename
+        message
+        path
+        minimumLength
+      }
+
+      # Interface contract to handle general purpose Error
+      ... on UserError {
+        message
+        path
+      }
+    }
+  }
+}
+```
+
+As you can see from this mutation input, the following mutation should result in the `EmailTaken` and `PasswordTooShort` Error based on our resolver.
+
+The UserError is to handle any general purpose error that way we get extensibility of the interface with the expressivity of the union.
+
+Hence, this mutation will return a response that looks like this.
+
+```gql
+
+data: {
+  signUpData: {
+      user: null,
```

Comment:
Again this is wrong at the schema level, you've defined the fields inside signUpData to be non nullable, but you're returning null for those fields here. Instead you should return null for the signUpData field as a whole.

Make sure the schema you use and the resolvers you implement map to each other correctly. Applies for all the examples you've used in this doc.